client/docs/service_groups.md

```diff
@@ -0,0 +1,160 @@
+locally. Another group can be defined to for final production
+deployment.
+
+Service groups can be easily defined for all of the services
+associated with a given site (see the various `create_from_site`
+subcommands below). They can also be created for services provided by
+multiple sites (for greater resiliency and availability).
+
+All service groups have a name and a list of services (generally
+represented by the names or URLs for the services defined in the
+service endpoint database). Each type of service group (`eservice`,
+`pservice`, and `sservice`) has additional unique attributes that
+describe policies that accompany the group. For example, enclave
+services have a preferred service that identifies a place more likely
+to have contract state pre-cached.
+
+While the name space for each type of service group is independent
+(that is, you can have an enclave service group with the same name as
+a provisioning service group), for convenience we often use the same
+name for groups of different types when we use them together. For
+example, we may name enclave, provisioning and storage service groups
+`develop` because we always use them together for contract
+development.
+
+While groups can be specified for any operation that creates a
+contract, if no group is specified, then a group called 'default' will
+be used. That group must be created.
+
+**NOTE:** Services included in service groups must be stored in the
+service endpoint database (see `pdo-service-db` command). Generally
+this means that service endpoint must be added (usually by importing a
+site configuration file) before groups can be created.
+
+## Service Groups Data File ##
+
+Service group information is stored in a database file identified by
+the configuration variable `Service.GroupDatabaseFile` which defaults
+to `${PDO_HOME}/data/groups_db.mdb`. Command line switches allow for
+the use of alternative group files.
+
+## Commands for Working with Service Groups ##
+
+### Service Groups: `pdo-service-groups` ###
+
+The `pdo-service-groups` command provides operations over the collection
+of services in the service groups database.
+
+* `clear` : remove all data from the service database
+* `import` : import service group definitions from a file
+* `export` : export service group definitions to a file
+* `list` : list service groups of a service type
+* `info` : provide detailed information about a group
+
+### Enclave Service Groups
+
+The `pdo-eservice` command provides operations to create, delete and
+modify groups of enclave services. In addition to a list of services,
+`eservice` groups include a preferred service that serves as a default
+location for operating on the contract. If the preferred service is
+`random`, then a different `eservice` will be chosen for each
+operation.
+
+* `create` : create a new service group
+* `create_from_site` : create a new service group using all of the services in a site file
```

Comment:
Wouldn't you want also an equivalent `add_from_site` to an existing group?

Reply:
that's exactly what this does.

Reply:
What i meant is `add_from_site_to_existing_group`.  `create_from_site` (according to comment) always creates a new one ...

Reply:
Ah, looking at code i realize this function does create new group only if it doesn't exist and otherwise uses existed one.  Correspondingly suggest changing the docu to make this more explicit
```suggestion
* `create_from_site` : add all of the services in a site file to a service group, creating it if it does not yet exist
```